job/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -272,6 +292,50 @@ private void copyLogs() {
             }
         }
     }
+
+    private String getLogPrefix(FlowNode node) {
+        BodyInvocationAction bodyInvocationAction = node.getAction(BodyInvocationAction.class);
+
+        if (bodyInvocationAction != null) {
+            BodyExecutionLabelAction bodyLabelAction = node.getAction(BodyExecutionLabelAction.class);
+
+            if (bodyLabelAction != null) {
+                return bodyLabelAction.getBodyLabel();
+            } else {
+                // This exec body does not have a body label. Stop now.
+                return null;
+            }
+        }
+
+        List<FlowNode> parents = node.getParents();
+        if (parents != null && !parents.isEmpty()) {
```

Comment:
I prefer having checks like that even when, in theory, they are not needed atm (that can change if someone changes the called code). That said, I'll remove it if you feel strongly about it.